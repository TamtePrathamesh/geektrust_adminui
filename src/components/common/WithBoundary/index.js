import React from 'react';

const DefaultError = () => <h1>Something went wrong.</h1>;

const WithBoundary = (Child, OnError = null) => {
  return class index extends React.Component {
    constructor(props) {
      super(props);
      this.state = { hasError: false };
    }
    componentDidCatch(error, errorInfo) {
      console.error(error, errorInfo);
      this.setState({ hasError: true });
    }
    render() {
      if (this.state.hasError) {
        return OnError ? <OnError /> : <DefaultError />;
      }
      return <Child {...this.props} />;
    }
  };
};

export default WithBoundary;
