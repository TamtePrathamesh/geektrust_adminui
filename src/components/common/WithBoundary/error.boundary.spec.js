import React from 'react';
import { render } from '@testing-library/react';
import WithBoundary from '.';

const ErrorComp = () => {
  throw new Error('Test comp');
};

const FineComp = () => 'Fine';

describe('withBoundary', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation();
    console.error.mockClear();
  });

  it('should match snapshot', () => {
    const Comp = WithBoundary(ErrorComp);
    expect(render(<Comp />).container).toMatchSnapshot();
    expect(console.error).toBeCalled();
  });

  it('should do nothing if there is no error', () => {
    const Comp = WithBoundary(FineComp);
    expect(render(<Comp />).container).toMatchSnapshot();
    expect(console.error).not.toBeCalled();
  });
});
