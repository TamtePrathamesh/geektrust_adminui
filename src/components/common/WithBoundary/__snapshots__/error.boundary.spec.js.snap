// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withBoundary should do nothing if there is no error 1`] = `
<div>
  Fine
</div>
`;

exports[`withBoundary should match snapshot 1`] = `
<div>
  <h1>
    Something went wrong.
  </h1>
</div>
`;
